CompilerSource/parsing/parser.cpp

```diff
@@ -1723,6 +1692,7 @@ std::unique_ptr<AST::Node> TryParseOperand() {
         TryParseTypeSpecifier(&type);
         if (token.type == TT_BEGINPARENTH) {
           auto tok = token;
+          token = lexer->ReadToken();
```

Comment:
We need to consume the token here because if we don't, `TryParseExpression` will treat it as a `PARENTHETICAL`,
 consume the `)` token, and the subsequent `require_token` will fail.

Reply:
Either is fine. It normally makes sense to let TryParseExpression handle it, except we've already decided that the expression is a functional cast here, so you're right to avoid letting it consume this token.

I suspect that this code is actually still wrong, though, as it seems to treat identifiers this same way. I think it's getting away with it right now because we don't have any tests for lambdas, which would mean that identifiers here should be treated as function calls.

Reply:
Great, I will take this into consideration. I will pay more attention to lambdas in the upcoming PR.